packages/node-terra/src/utils/terra-helper.ts

```diff
@@ -102,10 +107,18 @@ export function filterEvents(
 }
 
 async function getBlockByHeight(api: TerraClient, height: number) {
-  return api.blockInfo(height).catch((e) => {
+  let blockInfo: BlockInfo;
+  try {
+    if (api.mantlemintHealthOK) {
+      blockInfo = await api.blockInfoMantlemint(height);
+    } else {
+      blockInfo = await api.blockInfo(height);
+    }
```

Comment:
I think we can move this case to the TerraApi. This file shouldn't be concerned about the implementation that is used to fetch the content. We can do the same for TxInfo
 
```ts
// In TerraApi
async blockInfo(height) {
  if (this.mantlemintHealthOK) {
     return this.blockInfoMantlemint(height);
  }
  
  // return current implementation
}

```